src/ssl.c

```diff
@@ -32069,7 +32077,8 @@ int wolfSSL_ASN1_TIME_diff(int *days, int *secs, const WOLFSSL_ASN1_TIME *from,
     }
 
     fromSecs = XMKTIME(fromTm);
-    if (fromSecs < 0) {
+    /* Result can be negative due to time zones around UNIX epoch */
+    if (fromSecs == -1) {
```

Comment:
On POSIX, we can clear `errno` before calling `XMKTIME()`, and check `errno` for zeroness afterwards, to disambiguate error from actual `(time_t)-1`.  Same for the below `XMKTIME()` call.

Reply:
That works unless its a custom user defined `mktime`. Thoughts on that scenario?

Reply:
Although I suppose then that that wouldn't be a proper `mktime` implementation.

Reply:
Modified.

Reply:
Yes it does. `difftime` basically just subtracts the numbers and takes into account the possibility of `time_t` being signed. In my opinion this is a solid solution that will always work while also handling a possible error. https://github.com/bminor/glibc/blob/master/time/difftime.c

I originally thought about fixing this from the timezone side but I feel that messing with the system timezones or trying to use the `tm_gmtoff` member of `struct tm` will be much more unstable than just working with what the system returns.